src/WebJobs.Script.WebHost/Security/SecretManager.cs

```diff
@@ -40,16 +41,9 @@ public SecretManager(string secretsPath, IKeyValueConverterFactory keyValueConve
 
             Directory.CreateDirectory(_secretsPath);
 
-            _fileWatcher = new FileSystemWatcher(_secretsPath, "*.json")
-            {
-                IncludeSubdirectories = true,
-                EnableRaisingEvents = true
-            };
+            _fileWatcher = new AutoRecoveringFileSystemWatcher(_secretsPath, "*.json");
 
             _fileWatcher.Changed += OnChanged;
-            _fileWatcher.Created += OnChanged;
```

Comment:
With this change, I have simplified to API a bit so instantiating and subscribing to the different events requires less code and becomes a bit simpler/cleaner.